deblur/parallel_deblur.py

```diff
@@ -0,0 +1,71 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015, The Deblur Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+from __future__ import division
+
+from os.path import basename, join, split, splitext
+from os import mkdir
+
+from bfillings.sortmerna_v2 import build_database_sortmerna
+from skbio.parse.sequences import parse_fasta
+
+from qiime.parallel.util import ParallelWrapper
+
+
+class ParallelDeblur(ParallelWrapper):
```

Comment:
the memory for deblurring itself is <8Gb. Don't know about the other tools
in the pipeline
runtime for deblurring is ~1 min per sample i would guess...
Usually, when i run the old deblur pipeline for a miseq run, on 25 QSUBS in
parallel, it takes ~10 minutes (not including the split_libraries and split
on sample name)

On Mon, Feb 29, 2016 at 3:27 PM, Daniel McDonald <notifications@github.com>
wrote:

> In deblur/parallel_deblur.py
> <https://github.com/biocore/deblur/pull/29#discussion_r54498191>:
>
> > +# Distributed under the terms of the BSD 3-clause License.
> > +#
> > +# The full license is in the file LICENSE, distributed with this software.
> > +# ----------------------------------------------------------------------------
> > +from __future__ import division
> > +
> > +from os.path import basename, join, split, splitext
> > +from os import mkdir
> > +
> > +from bfillings.sortmerna_v2 import build_database_sortmerna
> > +from skbio.parse.sequences import parse_fasta
> > +
> > +from qiime.parallel.util import ParallelWrapper
> > +
> > +
> > +class ParallelDeblur(ParallelWrapper):
>
> What's the rough rough balltime runtime estimate per sample for, say,
> 150nt @ 10,000 seqs per sample? Are there large memory requirements like >
> 8GB per sample?
>
> We do per-sample parallelism on the AG data in the processing notebooks
> using multiprocessing so I think it's still relevant unless resource
> needs are prohibitive
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/deblur/pull/29/files#r54498191>.
>